script/testScripts/gov/ERC20TransferAction/ProposeERC20Transfer.s.sol

```diff
@@ -0,0 +1,85 @@
+contract ProposeERC20Transfer is JSONScript {
+  function run() public {
+    /// REQUIRED ENV VARS ///
+    address governanceAddress = vm.envAddress('GOVERNANCE_ADDRESS');
+    address ERC20TokenAddress = vm.envAddress('ERC20_TRANSFER_TOKEN_ADDRESS');
+    address receiverAddress = vm.envAddress('ERC20_TRANSFER_RECEIVER_ADDRESS');
+    address fromAddress = vm.envAddress('ERC20_TRANSFER_FROM_ADDRESS');
+    uint256 amountToTransfer = vm.envUint('ERC20_TRANSFER_AMOUNT');
+    /// END REQUIRED ENV VARS ///
+
+    ODGovernor gov = ODGovernor(payable(governanceAddress));
+    address[] memory targets = new address[](1);
+    {
+      targets[0] = ERC20TokenAddress;
+    }
+    // No values needed
+    uint256[] memory values = new uint256[](1);
+    {
+      values[0] = 0;
+    }
+
+    // Encode the calldata
+    bytes[] memory calldatas = new bytes[](1);
+    calldatas[0] = abi.encodeWithSelector(IERC20.transferFrom.selector, fromAddress, receiverAddress, amountToTransfer);
+
+    // Get the description and descriptionHash
+    string memory description = 'Transfer ERC20 tokens to receiver';
+    bytes32 descriptionHash = keccak256(bytes(description));
+
+    vm.startBroadcast(vm.envUint('GOV_EXECUTOR_PK'));
+
+    // Propose the action to add the collateral type
+    uint256 proposalId = gov.propose(targets, values, calldatas, description);
+    string memory stringProposalId = vm.toString(proposalId);
+
+    // Verify the proposalId is expected
+    assert(proposalId == gov.hashProposal(targets, values, calldatas, descriptionHash));
+
+    {
+      string memory ERC20TokenAddressAsString = vm.toString(ERC20TokenAddress);
+      string memory fromAddressAsString = vm.toString(fromAddress);
+      string memory receiverAddressAsString = vm.toString(receiverAddress);
+
+      // Build the JSON output
+      string memory objectKey = 'PROPOSE_ERC20_TRANSFER_KEY';
+      vm.serializeString(objectKey, 'tokenAddress', ERC20TokenAddressAsString);
+      vm.serializeString(objectKey, 'fromAddress', fromAddressAsString);
+      vm.serializeString(objectKey, 'toAddress', receiverAddressAsString);
+      vm.serializeUint(objectKey, 'amountToTransferInWei', amountToTransfer);
+      string memory jsonOutput =
+        _buildProposalParamsJSON(proposalId, objectKey, targets, values, calldatas, description, descriptionHash);
+      vm.writeJson(jsonOutput, string.concat('./gov-output/', stringProposalId, '-transfer-tokens-proposal.json'));
+    }
+
+    // Expected JSON output:
+    // {
+    //   "proposalId": uint256,
+    //   "tokenAddress": string,
+    //   "fromAddress": string,
+    //   "toAddress": string,
+    //   "amountToTransferInWei": string,
+    //   "targets": address[],
+    //   "values": uint256[]
+    //   "calldatas": bytes[],
+    //   "description": string,
+    //   "descriptionHash": bytes32,
```

Comment:
Great work. Do you have any suggestions for how delegates and community members can verify the contract call data in proposals? This is especially helpful for non solidity developers. We want everyone to be able to make informed decisions. For example, here is a recent HAI proposal on Tally, which I can't expect anyone other than seasoned web3 devs to interpret 

![image](https://github.com/open-dollar/od-contracts/assets/35622595/0953f236-cd03-4c06-baf1-53300164b704)


Here's the recent discussion in HAI's Discord on the subject:  https://discord.com/channels/1110925389464883201/1206632791513759825/1217556013151162410

Reply:
In web3 governance, we're tackling two big issues:

**Checking if the Code is Correct:** (_Valid Code_) We've got tools like Tally and Tenderly for this. Tally runs the code to see if it does what it's supposed to, which helps us trust the code more.
**Making Sure the Code Matches Its Goal:** (_Fit for propose_) This part's trickier. Sometimes, a proposal says it will do one thing, but the actual code does something different. It's vital that not just coders, but everyone, can understand what a proposal's code is really about.

To bridge the gap for non-developers, we can create a companion webpage for governance app. We have a predefine set of governance actions that are valid and fit for propose.

This page would help explain the actions proposed in a way that's easy to understand. If a proposal can't be explained by this page (are not from our action library), that's a signal for us to take a closer look before moving forward.

Process workflow could be something like: Automatically connect Tenderly's code-checking tool to our explanatory companion page. Whenever a proposal is made, Tenderly would check the code, and our page would translate what each part means in plain English. Proposals that can't be decoded this way would be flagged as "needs review."

This should be a broader discussion, I will copy this text and added to discord.